general/table.hpp

```diff
@@ -16,6 +16,8 @@
 
 #include "mem_alloc.hpp"
 #include "array.hpp"
+#include <ostream>
+#include <istream>
```

Comment:
These two headers are already included by `array.hpp`. Any reason to include them again here?

Reply:
since table.hpp uses the streams explicitly, I think these includes should be listed in table.hpp as well.